### PR TITLE
v1.4 beta !

### PR DESCRIPTION
**Pull request by Rurik19**
This PR solves a lot of issues, it improves users workflow by removing all api calls from interface to async backgroud. I think all issues authors should test it before release.
Issues solved:
#27 #28 #48 #49 #52 #59 #62 #63

### CODE REVIEW

**Comment by tcitworld** [comment]
Looks good.

Don't forget to create a beta release (and mark it as pre-release). :)

**Comment by Simounet** [comment]
It sounds great but it's kind of a mess into the commits' history. You didn't rebase the master branch and did the same job as the master branch. I'm trying to fix this but it is not that simple.

**Comment by Rurik19** [comment]
@Simounet  I dont't think you should waste your time on commit history. Let's just check the code and merge it when all is ok.

**Comment by Simounet** [comment]
Good job! Let me test that today. Thanks again Yuriy.

**Comment by Rurik19** [comment]
I have some thoughts about some improvement. I think there will be at least one commit before release

File: wallabagger/js/background.js [hunk 1]
```diff
@@ -1,207 +1,457 @@
...
-                requestExists(tab.url);
-            }
-        });
+    Port.onDisconnect.addListener(function () { portConnected = false; });
+    Port.onMessage.addListener(onPortMessage);
+}
+
+function addListeners () {
+    browser.contextMenus.onClicked.addListener(onContextMenusClicked);
+    browser.commands.onCommand.addListener(onCommandsCommand);
```
Comment (Simounet) [inline wallabagger/js/background.js]: I'm sorry but I don't get what's the difference between this custom handling and the basic one with `_execute_browser_action`? It seems to me that both doing the same thing.

Comment (Rurik19) [inline wallabagger/js/background.js]: @Simounet: Hi! I am back!
onCommand listener gets invoked when hotkey defined in manifest.json is pressed, while _execute_browser_action handles mouse click on icon.

Comment (Rurik19) [inline wallabagger/js/background.js]: Hi! I am back!
What do you mean of "basic" handling

Comment (Rurik19) [inline wallabagger/js/background.js]: _execute_browser_action is handled by browser itself
we have another shortcut for background saving - it have its own command "wallabag-it" - that we have to handle separately.

Comment (Simounet) [inline wallabagger/js/background.js]: Ok.

**Comment by Simounet** [comment]
I fixed the last couple of things that bothered me. I think this is ready to be merged. Is there anyone against it? 🍽

**Comment by ngosang** [comment]
Awesome work, thank you very much!

**Pull request merged**